### PR TITLE
Implement unit testing for the `cache` module

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
This PR add unit tests for the `cache.py` module.

### CODE REVIEW

File: coriolis/tests/test_cache.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from unittest import mock
+from coriolis import exception
+from coriolis import cache
+
+
+class TestGetCacheDecorator(unittest.TestCase):
+
+    @mock.patch.object(cache.cache, 'get_memoization_decorator')
+    def test_get_cache_decorator(self, mock_get_memoization_decorator):
+        mock_get_memoization_decorator.return_value = 'ExpectedValue'
```
Comment (Dany9966) [inline coriolis/tests/test_cache.py:14]: Just for future reference: you don't have to necessarily assign a value to `return_value` in this case. 
You can simply:
```python
self.assertEqual(result, mock_get_memoization_decorator.return_value)
```

File: coriolis/tests/test_cache.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_get_cache_decorator_invalid_provider_name(self):
+        invalid_providers = [123, '', None]
+
+        for provider in invalid_providers:
+            with self.assertRaises(exception.CoriolisException):
+                cache.get_cache_decorator(provider)
+
+
+if __name__ == '__main__':
```
Comment (Dany9966) [inline coriolis/tests/test_cache.py:33]: Please also remove these before posting a PR

File: coriolis/tests/test_cache.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import unittest
+from unittest import mock
+from coriolis import exception
+from coriolis import cache
+
+
+class TestGetCacheDecorator(unittest.TestCase):
```
Comment (Dany9966) [inline coriolis/tests/test_cache.py:10]: Base class of `TestGetCacheDecorator`, and all future test classes, should be `tests.test_base.CoriolisBaseTestCase`. Therefore, please:

```python
from tests import test_base
#...
class TestGetCacheDecorator(test_base.CoriolisBaseTestCase):
```

**Review by Dany9966 — CHANGES_REQUESTED** [review]
First review pass done. Mostly LGTM

File: coriolis/tests/test_cache.py (hunk 1 above)
Comment (mihaelabalutoiu) [reply coriolis/tests/test_cache.py]: Perfect! Good to know for the future! Thanks!

File: coriolis/tests/test_cache.py (hunk 3 above)
Comment (mihaelabalutoiu) [reply coriolis/tests/test_cache.py]: Good to know! Thanks a lot!

**Comment by mihaelabalutoiu** [comment]
I made the changes, please take a look now :) @Dany9966

File: coriolis/tests/test_cache.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+from coriolis import exception
+from coriolis import cache
+from coriolis.tests import test_base
+
+
+class TestGetCacheDecoratorTestCase(test_base.CoriolisBaseTestCase):
```
Comment (Dany9966) [inline coriolis/tests/test_cache.py:10]: NIT: Class name should simply be `CacheTestCase`, as we should add future methods to test in this class (not just `get_cache_decorator` specific). Please rename it

**Review by Dany9966 — COMMENTED** [review]
Only one small change left.

File: coriolis/tests/test_cache.py (hunk 4 above)
Comment (mihaelabalutoiu) [reply coriolis/tests/test_cache.py]: Done, thanks a lot for everythink! :)

**Pull request merged**